architecture.md

```diff
@@ -0,0 +1,91 @@
+For example, a user could write a system that applies force to an entity by setting a component on that entity, and the physics system would pick that up and react by applying the force to make the entity actually move.
+The modification of entities and components is how server plugins communicate with each other.
+Other Ignition Libraries, however, don’t know about the entities and components.
+
+There is a loop running in the backend that runs the systems, where some systems are proposing changes to entities and their components, and other systems are handling and applying those changes.
+This is called the “simulation loop”.
+The Entity Component Manager (ECM) in the backend provides the functionality for the actual querying and updating of the entities and components.
+
+Physics, user commands, and scene broadcaster are all systems launched by default in the backend.
+As mentioned earlier, however, even default systems can be added to or removed from this loop.
+For any other functionality, like sensor data processing for example, an additional system would have to be loaded.
+For example, if you need to generate sensor data that utilizes rendering sensors, you would need to load the sensors system.
+The visualization of that data, however, is left up to the client side plugins.
+
+## Communication process
+
+Any information that crosses the process boundary (whether back-to-front with scene broadcaster or front-to-back with user commands) has to go through [Ignition Transport](https://ignitionrobotics.org/libs/transport), the communication library.
+Synchronization between processes is performed by scene broadcaster, a server-side Ignition Gazebo plugin that uses Ignition Transport and [Ignition Messages](https://ignitionrobotics.org/libs/msgs) libraries to send messages from the server to the client.
+The messages themselves are provided by Ignition Messages, while the framework for creating the publishers and subscribers that exchange messages is from Ignition Transport.
+
+The scene broadcaster system is responsible for getting the state of the world (all of the entities and their component values) from the ever-changing simulation loop running in the back end, packaging that information into a very compact message, and periodically sending it to the frontend to process.
+Omitting the scene broadcaster, while possible, would mean not being able to visualize data on the frontend, which can be useful for saving computational power.
+
+In addition to state messages sent from the scene broadcaster to the client, the client can also make requests to the server using the user commands system (another backend plugin).
+It also utilizes the Ignition Transport and Ignition Messages libraries.
+The user command system has services that can be requested by the frontend GUI to insert, delete, move, etc., and will send responses back to the GUI acknowledging the receipt and execution of those requests.
```

Comment:
Should we say what these "operations" (insert, delete, move, etc.) are being performed on? Are they being performed on models, or is there a chance that they can be performed on other things, too?

Reply:
Models, lights (and eventually actors)